docs/src/main/asciidoc/config-yaml.adoc

```diff
@@ -109,55 +206,116 @@ app:
               plain-text: true
 ----
 
-== Profiles
+// CREATE include::modules/proc_using-configuration-profiles-with-yaml.adoc[leveloffset=+2]
+[id="proc_using-configuration-profiles-with-yaml_{context}"]
+== Using configuration profiles with YAML
 
-As you can see in the previous snippet, you can use xref:config-reference.adoc#profiles[profiles] in YAML. The profile
-key requires double quotes: `"%test"`. This is because YAML does not support keys starting with `%`.
+You can use xref:config-reference.adoc#profiles[profiles] in YAML.
 
-Everything under the `"%test"` key is only enabled when the `test` profile is active. For example, in the previous
-snippet it disables OIDC (`quarkus.oidc.enabled: false`), whereas without the `test` profile, it would be enabled.
+Because YAML does not support names that start with `%`, you must enclose the names of the profiles' configuration keys between double quotes.
 
-As for the Java Properties format, you can define your own profile:
+For example, in `application.yaml`, you must use `"%dev"` or `"%test"` as the configuration keys to identify properties that you want your Quarkus project to use when you build and run it using the development or test profiles.
 
+.Example `application.yaml` with `prod` and `dev` profiles
 [source, yaml]
 ----
 quarkus:
   http:
     port: 8081
 
-"%staging":
+"%dev":
     quarkus:
         http:
           port: 8082
 ----
 
-If you enable the `staging` profile, the HTTP port will be 8082, whereas it would be 8081 otherwise.
+With the preceding example, if you don't specify a profile, Quarkus runs the `prod` profile by default, and the HTTP port number is `8081`.
+If you enable the `dev` profile, the HTTP port number is `8082`.
```

Comment:
Please don't change the meaning of this example. The dev profile is used in dev mode. Prod is not always the default and has a very special meaning in Quarkus.
Also the default port is valid for all profiles if not specified otherwise so what you have written not really correct IMO.